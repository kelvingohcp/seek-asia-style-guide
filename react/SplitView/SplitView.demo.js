import React from 'react';
import SplitView from './SplitView';
import { Card, Section, Text } from 'seek-asia-style-guide/react';

const leftCards = number => {
  const cards = [];
  for (let i = 0; i < number; i++) {
    cards.push(
      <Card>
        <Section>
          <Text shouting>Left Content</Text>
          <Text>
            This content is provided by the first child. It will take the whole
            container on mobile and tablet.It has 2 different sizes on desktop
            and large screen.
          </Text>
        </Section>
      </Card>
    );
  }
  return <div key="leftCards">{cards}</div>;
};

const rightCards = number => {
  const cards = [];
  for (let i = 0; i < number; i++) {
    cards.push(
      <Card>
        <Section>
          <Text shouting>Right Content</Text>
          <Text>
            This content is provided by the second child. It will not show on
            mobile or tablet.
          </Text>
        </Section>
      </Card>
    );
  }
  return <div key="rightCards">{cards}</div>;
};

export default {
  route: '/split-view',
  title: 'Split View',
  component: SplitView,
  initialProps: {
    children: [leftCards(3), rightCards(5)]
  },
  options: [
    {
      label: 'Height',
      type: 'radio',
      states: [
        {
          label: 'No height',
          transformProps: props => ({
            ...props
          })
        },
        {
          label: '300px',
          transformProps: props => ({
            ...props,
            height: 300
          })
        },
        {
          label: '500px',
          transformProps: props => ({
            ...props,
            height: 500
          })
        }
      ]
    }
  ]
};
