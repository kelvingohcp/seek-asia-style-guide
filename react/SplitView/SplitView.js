import React from 'react';
import styles from './SplitView.less';
import PropTypes from 'prop-types';

export default function SplitView({ children }) {
  return (
    <div className={styles.root}>
      {children &&
        children.length > 0 && (
        <div className={styles.leftChild}>{children[0]}</div>
      )}
      {children &&
        children.length > 1 && (
        <div className={styles.rightChild}>{children.slice(1)}</div>
      )}
    </div>
  );
}

SplitView.propTypes = {
  children: PropTypes.node.isRequired
};
