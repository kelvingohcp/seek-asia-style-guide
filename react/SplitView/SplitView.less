@import (reference) '~seek-asia-style-guide/theme';

.root {
  display: flex;
}

.leftChild {
  flex: 1;
  @media @new-tablet, @new-desktop {
    max-width: @gutter-width * 16;
  }
  @media @new-desktop-widescreen {
    max-width: @gutter-width * 22;
  }
  overflow-y: auto;
}

.rightChild {
  display: none;
  flex: 1;
  @media @new-tablet, @new-desktop, @new-desktop-widescreen {
    display: inherit;
  }
  overflow-y: auto;
}
