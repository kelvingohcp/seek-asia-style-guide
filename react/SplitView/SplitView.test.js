import React from 'react';
import { shallow } from 'enzyme';

import SplitView from './SplitView';

const renderSplitView = props =>
  shallow(
    <SplitView {...props}>
      <div> Child 1</div>
      <div> Child 2</div>
    </SplitView>
  );

describe('Split View:', () => {
  it('should render with default props', () => {
    expect(renderSplitView()).toMatchSnapshot();
  });

  it('should render with height prop', () => {
    expect(renderSplitView({ height: 200 })).toMatchSnapshot();
  });
});
