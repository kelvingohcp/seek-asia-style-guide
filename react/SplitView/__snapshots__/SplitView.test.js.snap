// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Split View: should render with default props 1`] = `
<div
  className="root"
>
  <div
    className="leftChild"
    style={
      Object {
        "height": undefined,
      }
    }
  >
    <div>
       Child 1
    </div>
  </div>
  <div
    className="rightChild"
    style={
      Object {
        "height": undefined,
      }
    }
  >
    <div>
       Child 2
    </div>
  </div>
</div>
`;

exports[`Split View: should render with height prop 1`] = `
<div
  className="root"
>
  <div
    className="leftChild"
    style={
      Object {
        "height": 200,
      }
    }
  >
    <div>
       Child 1
    </div>
  </div>
  <div
    className="rightChild"
    style={
      Object {
        "height": 200,
      }
    }
  >
    <div>
       Child 2
    </div>
  </div>
</div>
`;
